### PR TITLE
Feature/ticket 7/purchase presentation layer

### PR DESCRIPTION
**Pull request by luiggy-mamani-jalafund**
# US-07: Purchase - Presentation Layer

 ## What did I do it?
- Implemented purchase form in the presentation layer

## How did I do it?
- Implemented Popup
- Implemented purchase fields as molecules and their styles
- Implemented purchase form  using the popup and fields modules

## Why did I do it?
To help in the purchase user history.

image

### CODE REVIEW

**Comment by luiggy-mamani-jalafund** [comment]
[ID-07: Purchase](https://trello.com/c/wCEnN67I/12-id-07-purchase)

File: web-frontend/src/app/movies/[id]/page.js [hunk 1]
```diff
@@ -2,8 +2,9 @@ import MovieDetails from "@/components/template/movies/MovieDetails";
 import { fetchSingleMovie, fetchMovieShowtimes } from "@/utils/data_fetchers/MoviesFetcher";
 
 export default async function MoviePage({ params }) {
-    const movie = await fetchSingleMovie(params);
-    const showtimes = await fetchMovieShowtimes(params);
+    const movie = await fetchSingleMovie(params.id);
+    const showtimes = await fetchMovieShowtimes(params.id);
+    console.log(showtimes);
```
Comment (salet-gutierrez-jalafund) [inline web-frontend/src/app/movies/[id]/page.js]: remove this impression

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

File: web-frontend/src/components/template/theaters/Theater.jsx [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-                const decodedSchedule = decodeURIComponent(scheduleParam);
-                const parsedSchedule = JSON.parse(decodedSchedule);
-                setSchedule(parsedSchedule);
-            } catch (_) {
+        fetchMovieShowtimes(movieId)
+            .then((res) => {
+                for (let i = 0; i < res.length; i++) {
+                    for (let j = 0; j < res[i].hours.length; j++) {
+                        if (res[i].hours[j].id === hourId) {
+                            console.log(res[i].hours[j]);
```
Comment (salet-gutierrez-jalafund) [inline web-frontend/src/components/template/theaters/Theater.jsx]: delete console.log

File: web-frontend/src/utils/data_fetchers/MoviesFetcher.js [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
+    } catch (e) {
+        console.log(e);
 
-    return movies;
+        return [];
+    }
 }
 
 export async function fetchMovieShowtimes(movieId) {
+    console.log(`http://localhost:5011/api/Showtime/${movieId}`);
```
Comment (salet-gutierrez-jalafund) [inline web-frontend/src/utils/data_fetchers/MoviesFetcher.js]: delete console.log

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

**Review by salet-gutierrez-jalafund — APPROVED** [review]

**Pull request merged**